lib/puppet/util/execution.rb

```diff
@@ -336,6 +338,11 @@ def self.execute_posix(command, options, stdin, stdout, stderr)
       # turning "/tmp/foo;\r\n /bin/echo" into ["/tmp/foo;\r\n", " /bin/echo"]
       command = [command].flatten
       Process.setsid
+      if options[:priority]
+        if Process.uid == 0
+          Process.setpriority(Process::PRIO_PROCESS, 0, options[:priority])
```

Comment:
Could call [Puppet::Util::Limits.setpriority](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/limits.rb#L3) here, as it will handle the case where the underlying ruby wasn't built with `HAVE_GETPRIORITY`.

Reply:
Also could use `Puppet::Settings::PrioritySetting::PRIORITY_MAP[:normal]` to get the platform-specific priority value for windows or *nix.

Reply:
fixed, thanks